src/plugins/wizard/public/application/utils/get_top_nav_config.tsx

```diff
@@ -41,17 +41,24 @@ import { WizardVisSavedObject } from '../../types';
 import { StyleState, VisualizationState, AppDispatch } from './state_management';
 import { EDIT_PATH } from '../../../common';
 import { setEditorState } from './state_management/metadata_slice';
+import { EmbeddableStateTransfer } from '../../../../embeddable/public';
 interface TopNavConfigParams {
   visualizationIdFromUrl: string;
   savedWizardVis: WizardVisSavedObject;
   visualizationState: VisualizationState;
   styleState: StyleState;
   saveDisabledReason?: string;
   dispatch: AppDispatch;
+  originatingApp?: string;
+  setOriginatingApp?: (originatingApp: string | undefined) => void;
+  stateTransfer: EmbeddableStateTransfer;
 }
 
 export const getTopNavConfig = (
   {
+    originatingApp,
```

Comment:
Since the only place that we actually need the value for `originatingApp` is within the `getTopNavConfig` function, cant we skip the use state in `top_nav.tsx` and just fetch its value here since we already have access to `services` in this function?

That way we can also avoid passing all these props into the function

Reply:
Put originatingApp and stateTransfer within getTopNavConfig function. Had to change useMemo() hook to useEffect() in topNav to avoid "can not update during an existing state transition" error.